### PR TITLE
[openvdb] update to 12.0.0 and fix link error

### LINKED ISSUE #41951: [openvdb] update to 12.0.0
### Library name

openvdb

### New version number

12.0.0

### Other information that may be useful (release notes, etc...)

https://github.com/AcademySoftwareFoundation/openvdb/releases/tag/v12.0.0  
https://www.openvdb.org/documentation/doxygen/changes.html#v12_0_0_changes

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fixes #41951
Update openvdb to version 12.0.0
Fix link error and config error
Feature nanovdb and tools tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

Usage tested successfully by `openvdb:x64-windows`

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/openvdb/fix-build-error.patch [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ endif()
+ 
+ if(NANOVDB_USE_TBB AND NOT TARGET TBB::tbb)
+-  find_package(TBB ${MINIMUM_TBB_VERSION} REQUIRED)
++  find_package(TBB CONFIG REQUIRED)
+ endif()
+ 
+ if(NANOVDB_USE_BLOSC AND NOT TARGET Blosc::blosc)
+-  find_package(Blosc ${MINIMUM_BLOSC_VERSION} REQUIRED NAMES blosc)
++  find_package(blosc CONFIG REQUIRED)
```
Comment (dg0yt) [inline ports/openvdb/fix-build-error.patch:15]: This is patching the result of the other patch.

Comment (JonLiu1993) [reply ports/openvdb/fix-build-error.patch]: Done

**Comment by dg0yt** [comment]
The patches still do not treat separate topics. Maybe it is easier if all three patches are combined.

File: ports/openvdb/fix_cmake.patch [hunk 2]
```diff
@@ -59,10 +91,21 @@ index 33d4e5cc..63699b2c 100644
...
+-    if(${Log4cplus_VERSION} VERSION_LESS FUTURE_MINIMUM_LOG4CPLUS_VERSION)
+-      message(DEPRECATION "Support for Log4cplus versions < ${FUTURE_MINIMUM_LOG4CPLUS_VERSION} "
+-        "is deprecated and will be removed.")
+-    endif()
+-  endif()
+-endif()
+-
+-if(USE_BLOSC)
+-  # Find Blosc libraries
 -  find_package(Blosc ${MINIMUM_BLOSC_VERSION} REQUIRED)
```
Comment (dg0yt) [inline ports/openvdb/fix_cmake.patch:109]: IMO this block should be retained as much as possible (upstream code!).
~~~cmake
if(USE_BLOSC)
  # Find Blosc libraries
  find_package(Blosc NAMES blosc REQUIRED)
  if(TARGET blosc_shared)
    add_library(Blosc::blosc ALIAS blosc_shared)
  else()
    add_library(Blosc::blosc ALIAS blosc_static)
  endif()
~~~
and get rid of blosc changes in other parts of the file.

File: ports/openvdb/fix_cmake.patch [hunk 3]
```diff
@@ -59,10 +91,21 @@ index 33d4e5cc..63699b2c 100644
...
+ if(USE_LOG4CPLUS)
+   # Find Log4CPlus libraries
+-  find_package(Log4cplus ${MINIMUM_LOG4CPLUS_VERSION} REQUIRED)
+-  if(OPENVDB_FUTURE_DEPRECATION AND FUTURE_MINIMUM_LOG4CPLUS_VERSION)
+-    if(${Log4cplus_VERSION} VERSION_LESS FUTURE_MINIMUM_LOG4CPLUS_VERSION)
+-      message(DEPRECATION "Support for Log4cplus versions < ${FUTURE_MINIMUM_LOG4CPLUS_VERSION} "
+-        "is deprecated and will be removed.")
+-    endif()
+-  endif()
+-endif()
```
Comment (dg0yt) [inline ports/openvdb/fix_cmake.patch:105]: Set `USE_LOG4CPLUS` via an option if possible, or replae with `if(0)`.

**Review by dg0yt — COMMENTED** [review]
Now that we see the whole picture, we can finally talk how to trim the patch to what is really needed.

Getting started...

File: ports/openvdb/fix_cmake.patch (hunk 3 above)
Comment (JonLiu1993) [reply ports/openvdb/fix_cmake.patch]: Done

File: ports/openvdb/fix_cmake.patch (hunk 2 above)
Comment (JonLiu1993) [reply ports/openvdb/fix_cmake.patch]: Done

**Pull request merged**